astropy/coordinates/tests/test_geodetic_representations.py

```diff
@@ -2,16 +2,65 @@
 
 """Test geodetic representations"""
 import pytest
-from numpy.testing import assert_array_equal
 
+from numpy.testing import assert_array_equal
 from astropy import units as u
+from astropy.units import allclose as quantity_allclose
+from astropy.coordinates.geodetic import BaseGeodeticRepresentation
+
 from astropy.coordinates.earth import (
     GRS80GeodeticRepresentation,
     WGS72GeodeticRepresentation,
     WGS84GeodeticRepresentation,
 )
 from astropy.coordinates.representation import CartesianRepresentation
-from astropy.units import allclose as quantity_allclose
+from astropy.units.tests.test_quantity_erfa_ufuncs import vvd
+
+
+def test_custombasegeodetic_error():
+    # Test incomplete initialization.
+    msg = (
+        "type object 'BaseGeodeticRepresentation' has no attribute '_equatorial_radius'"
+    )
+    s1 = CartesianRepresentation(
+        x=[1, 3000.0] * u.km, y=[7000.0, 4.0] * u.km, z=[5.0, 6000.0] * u.km
+    )
+
+    custom_geodetic = BaseGeodeticRepresentation
+    custom_geodetic._flattening = 0.075 * u.dimensionless_unscaled
+    with pytest.raises(AttributeError, match=msg):
+        s2 = custom_geodetic.from_representation(s1)
+
+
+def test_cartesian_custombasegeodetic_roundtrip():
+    # Test array-valued input in the process.
+    s1 = CartesianRepresentation(
+        x=[1, 3000.0] * u.km, y=[7000.0, 4.0] * u.km, z=[5.0, 6000.0] * u.km
+    )
+
+    custom_geodetic = BaseGeodeticRepresentation
+    custom_geodetic._flattening = 0.075 * u.dimensionless_unscaled
+    custom_geodetic._equatorial_radius = 3000000.0 * u.m
+    s2 = custom_geodetic.from_representation(s1)
+
+    s3 = CartesianRepresentation.from_representation(s2)
+
+    s4 = custom_geodetic.from_representation(s3)
+
+    assert quantity_allclose(s1.x, s3.x)
+    assert quantity_allclose(s1.y, s3.y)
+    assert quantity_allclose(s1.z, s3.z)
+
+    assert quantity_allclose(s2.lon, s4.lon)
+    assert quantity_allclose(s2.lat, s4.lat)
+    assert quantity_allclose(s2.height, s4.height)
+
+    # Test initializer just for the sake of it.
```

Comment:
So, I think I would *start* with this, but use explicit lon, lat, height.

Reply:
Answering to this comment and the previous one.
I have added round-trip tests from cartesian and from geodetic representation, each with its own initialization.